ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +242,21 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0 && field->decimals < 7) {
```

Comment:
Why ignoring fraction with higher precision?

Reply:
Currently it is declared that the precision is microseconds at most; according to the manual, higher precision is not possible.
This is only to safeguard against possible future hard-to-find issues should any behavioral changes go unnoticed.